dart/collision/fcl/FCLCollisionDetector.h

```diff
@@ -38,50 +38,104 @@
 #ifndef DART_COLLISION_FCL_FCLCOLLISIONDETECTOR_H_
 #define DART_COLLISION_FCL_FCLCOLLISIONDETECTOR_H_
 
+#include <vector>
 #include <fcl/collision_object.h>
-#include <fcl/collision_data.h>
-#include <fcl/broadphase/broadphase.h>
-
 #include "dart/collision/CollisionDetector.h"
 
 namespace dart {
 namespace collision {
 
-class FCLCollisionNode;
+class FCLCollisionObject;
 
-/// FCLCollisionDetector
 class FCLCollisionDetector : public CollisionDetector
 {
 public:
-  /// Constructor
-  FCLCollisionDetector();
 
-  /// Destructor
+  static std::shared_ptr<FCLCollisionDetector> create();
+
+  /// Constructor
   virtual ~FCLCollisionDetector();
 
+  /// Return engine type "FCL"
+  static const std::string& getTypeStatic();
+
   // Documentation inherited
-  virtual bool detectCollision(bool _checkAllCollisions,
-                               bool _calculateContactPoints) override;
+  const std::string& getType() const override;
 
   // Documentation inherited
-  virtual CollisionNode* createCollisionNode(dynamics::BodyNode* _bodyNode)
-  override;
+  std::shared_ptr<CollisionGroup> createCollisionGroup() override;
 
-  /// Get collision node given FCL collision geometry
-  CollisionNode* findCollisionNode(
-      const fcl::CollisionGeometry* _fclCollGeom) const;
+  // Documentation inherited
+  bool detect(CollisionGroup* group,
+              const Option& option, Result& result) override;
 
-  /// Get collision node given FCL collision object
-  FCLCollisionNode* findCollisionNode(
-      const fcl::CollisionObject* _fclCollObj) const;
+  // Documentation inherited
+  bool detect(CollisionGroup* group1, CollisionGroup* group2,
+              const Option& option, Result& result) override;
+
+  /// FCL's primitive shape support is still not complete. FCL 0.4.0 fixed lots
+  /// of the bugs, but it still returns single contact point per shape pair
+  /// except for box-box collision. For this reason, we recommend using mesh for
+  /// primitive shapes until FCL fully support primitive shapes.
+  enum PrimitiveShape_t
```

Comment:
Will update the comment.

I thought using `_t` suffix is our convention as the recently added enum have the suffix: [1](https://github.com/dartsim/dart/blob/e8b4f1c40e26140d93ea803bcb4a76547ebe05fb/dart/dynamics/Chain.h#L84), [2](https://github.com/dartsim/dart/blob/ba23a8ae30e4e6148a91a813c18b4f5e566c9839/dart/common/Empty.h#L49), [3](https://github.com/dartsim/dart/blob/e14110c8f94cef2cfd93d3d4dc054ba8a8cb2a6b/dart/constraint/BalanceConstraint.h#L85), [4](https://github.com/dartsim/dart/blob/71f6aa2f8644138e321fe6bdfbffda6bcbf16289/dart/dynamics/InverseKinematics.h#L620).